sympy/physics/quantum/tests/test_tensorproduct.py

```diff
@@ -51,8 +51,8 @@ def test_tensor_product_simp():
 
 
 def test_tensor_product_simp_Pow():
-    assert tensor_product_simp((TP(A, B)*TP(C, D)) ** 2) == (TP(A*C, B*D)) ** 2
-    assert tensor_product_simp(TP(A, B) ** 2) == TP(A**2, B**2)
+    assert tensor_product_simp_Pow((TP(A, B)*TP(C, D)) ** 2) == (TP(A*C, B*D)) ** 2
```

Comment:
Should `(TP(A*C, B*D)) ** 2` be simplified in the same way as `TP(A, B) ** 2` on the next line?

Reply:
@jksuom Optimally, yes. As per my previous comment in the previous review-thread for this file, simplified expression can still be reached by applying `tensor_product_simp_Pow` again. I would say this test should either be removed, or improved.

The test (or a form of this test) that *needs* to be added is
```
assert tensor_product_simp_Pow(-1*(TP(A,B)**2)) == -1*TP(A**2, B**2)
```
which this PR would currently fail. See my last main-thread comment for details.

Reply:
I think we can raise an issue regarding the negative tensor product issue, and then solve it seperately. As of now, are the tests ok?

Reply:
To me, the current fix only fixes a very special case: when you have a `Pow` all alone. My "negative" is just an example for a more general expression, `a*TP(A, B)**2` won't be simplified either (for some symbol `a`). So I agree with @jksuom that more tests are needed.

I'm sorry I was not clear with this in the issue formulation (I only had a simple example as a proof of concept), but without those kind of tests being passable, the issue I'm having is not resolved. That is my opinion.

Reply:
Tests for different types of base expressions are needed. Currently, for example,
```
>>> T = TensorProduct(A, B)
>>> U = TensorProduct(C, D)
>>> tensor_product_simp_Pow((T + U)**2)
(AxB**2)x(CxD**2)
```
which is not correct.

Reply:
Great...
I ''ll be working on it..